### DISCUSSION TITLE
[License]: Is it possible to installl and use minio without modification of minio sources for my commercial data lakehouse platform?

### DISCUSSION THREAD

**Discussion by cloudcheflabs** [minio/minio#17441] [Q&A] [answered]
I want to use mino as s3 compatible object storage for my commercial data lakehouse platform for disconnected environment.
Minio will be installed without modification of minio source for my data lakehouse platform.

I am concerning AGPL v3 license of minio. 
Is it possible to use minio without minio sources for my commercial platform?

- Kidong.

**Comment by ravindk89** [comment, marked as answer]
Hi @cloudcheflabs 

We have designed MinIO as an Open Source software for the Open Source software community. This requires applications to consider whether their usage of MinIO is in compliance with the GNU AGPLv3 license.

MinIO cannot make the determination as to whether your application's usage of MinIO Server is in compliance with the AGPLv3 license requirements. You should instead rely on your own legal counsel or licensing specialists to audit and ensure your application is in compliance with the licenses of MinIO and all other open-source projects with which your application integrates or interacts.

We understand that AGPLv3 licensing is complex and nuanced. It is for that reason we strongly encourage using experts in licensing to make any such determinations around compliance instead of relying on apocryphal or anecdotal advice.

MinIO Commercial Licensing is the best option for applications which trigger AGPLv3 obligations (e.g. open sourcing your application). Applications using MinIO - or any other OSS-licensed code - without validating their usage do so at their own risk.

**Reply by ravindk89** [reply]
To add - whether or not your application is commercial or not, airgapped or not, etc. is not really what matters. What matters is compliance with AGPLv3, and that is fact-specific and may require consideration of your regional laws/regulations on the topic. So we really cannot give you strict advice.
